main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala

```diff
@@ -70,12 +71,26 @@ object EntryPoint {
           entryPoint =>
             root.copy(
               defs = root.defs + (entryPoint.sym -> entryPoint),
-              entryPoint = Some(entryPoint.sym)
+              entryPoint = Some(entryPoint.sym),
+              reachable = getReachable(root) + entryPoint.sym
             )
         }
       // Case 2: No entry point. Don't touch anything.
-      case None => Validation.success(root)
+      case None => Validation.success(root.copy(reachable = getReachable(root)))
+    }
+  }
+
+  /**
+   * Returns all reachable definitions.
+   */
+  private def getReachable(root: TypedAst.Root): Set[Symbol.DefnSym] = {
```

Comment:
Seems like this could easily be done functionally, but OK.

Reply:
I HAVE BECOME A FAN OF REGIONS

Reply:
Argh, caps.

Reply:
Yeah, its a bit lazy. I guess a filter could have done the job. But I have a map, so I would need a fold I guess.